Configurations.md

```diff
@@ -0,0 +1,1771 @@
+# Configuring Rustfmt
+
+Rustfmt is designed to be very configurable. You can create a TOML file called `rustfmt.toml` or `.rustfmt.toml`, place it in the project or any other parent directory and it will apply the options in that file.
+
+A possible content of `rustfmt.toml` or `.rustfmt.toml` might look like this:
+
+```toml
+array_layout = "Block"
+array_width = 80
+reorder_imported_names = true
+```
+
+# Configuration Options
+
+Below you find a detailed visual guide on all the supported configuration options of rustfmt:
+
+## `array_layout`
+
+Indent on arrays
+
+- **Default value**: `"Visual"`
+- **Possible values**: `"Block"`, `"Visual"`
+
+#### `"Block"`:
+
+```rust
+let lorem = vec![
+    "ipsum",
+    "dolor",
+    "sit",
+    "amet",
+    "consectetur",
+    "adipiscing",
+    "elit",
+];
+```
+
+#### `"Visual"`:
+
+```rust
+let lorem = vec!["ipsum",
+                 "dolor",
+                 "sit",
+                 "amet",
+                 "consectetur",
+                 "adipiscing",
+                 "elit"];
+```
+
+## `array_width`
+
+Maximum width of an array literal before falling back to vertical formatting
+
+- **Default value**: `60`
+- **Possible values**: any positive integer
```

Comment:
to clarify, in other places it will mean other things, but usually it means always or never something which some people want